python/ray/train/data_parallel_trainer.py

```diff
@@ -427,7 +428,19 @@ def _report(self, training_iterator: TrainingIterator) -> None:
         for results in training_iterator:
             # TODO(ml-team): add ability to report results from multiple workers.
             first_worker_results = results[0]
-            tune.report(**first_worker_results)
+            if _use_storage_context():
+                assert (
+                    isinstance(first_worker_results, tuple)
+                    and len(first_worker_results) == 2
+                )
+                metrics, checkpoint = first_worker_results
+                logger.debug(
+                    "Report (metrics, checkpoint) to the Tune session:\n"
+                    f"  metrics={metrics}\n  checkpoint={checkpoint}"
+                )
+                train.report(metrics, checkpoint=checkpoint)
```

Comment:
Btw, a way to make this perhaps easier to read is to not use the public `report` API for this inner reporting, but call a private `_report` that doesn't do the repeated checkpoint uploading.

However, this is a minor comment.

Reply:
@ericl I think that makes sense -- I think this will be easier to do when we unify the 2 sessions -- just introduce a private method on the session that does the Train -> Tune communication.

Let's merge with what we have for now

Reply:
Actually, hold off on that, need to rebase and up the test timeout limit.